docs/ruleset_functions/compare_swh_dist_systems_and_components.md

```diff
@@ -0,0 +1,54 @@
+## compare_swh_dist_systems_and_components
+
+Description: This function compares all sub-components of a given SWH Distribution System ID between two models.  The function also accepts   
+
+Inputs:
+- **RMD1**
+- **RMD2**
+- **compare_context_str**
+- **swh_distribution_id**
+
+Returns:
+- **errors**: An array of errors encountered in this function and in compare_context_pair.  If errors is an empty array, then all elements that were expected to match do match
+
+Function Call:
+
+- **compare_context_pair** - there is no RDS for this function, but it is a function developed for Rule 1-6 that compares two elements  
+- **get_component_by_id**
+
+Data Lookup: None
+
+Logic:
+  - create the list errors: `errors = []`
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD1_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD1)`
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD2_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD2)`
+
+  - check if this distribution ID is in model 1: `if RMD1_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD1: `RMD1_swh_equipment_dict = RMD1_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD1_swh_distribution = get_component_by_id(RMD1, swh_distribution_id)`
+  - otherwise, return FALSE by adding an error to the list: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"`
+
+  - check if this distribution ID is in model 2: `if RMD2_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD2: `RMD2_swh_equipment_dict = RMD2_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD2_swh_distribution = get_component_by_id(RMD2, swh_distribution_id)`
+  - otherwise, return FALSE by adding an error to the list: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"`
+
+
```

Comment:
if the list of errors is not empty at this point I think we should return it and prevent the rest of the logic

Reply:
It's a preference, but I like to provide as many errors as possible to the user at once so they can resolve them at the same time - reduces iterations for the modeler

Reply:
My take is if the `swh_distribution_id` is not found in one or another, continue processing can become confusing since the rest checks are depending on having these two matched distribution objects.

It would be good to enforce the same length of `RMD1_swh_distribution` and `RMD2_swh_distribution` lists. For example, adding an assert to make sure these two are in the same length. For ones that mismatched (not exist), we can add to the error list, and skip adding them to the `RMD1_swh_distribution` or `RMD2_swh_distribution` since further comparison is just comparing value with `None` anyway.